### PR TITLE
Set monitoring by config

### PR DESCRIPTION
**Pull request by AmitKarin**
created a new func change_component_monitoring_mode that sets the monitoring mode for the component.

### CODE REVIEW

**Comment by ItayHoresh** [comment]
@AmitKarin please check why the travis build checks is failing.

**Comment by AmitKarin** [comment]
@ItayHoresh all checks are done. can you approve?

File: pipert/core/component.py [hunk 1]
```diff
@@ -1,6 +1,9 @@
 import threading
 from prometheus_client import start_http_server
 from torch.multiprocessing import Event, Process
+
+from pipert.contrib.metrics_collectors.prometheus_collector import PrometheusCollector
+from pipert.contrib.metrics_collectors.splunk_collector import SplunkCollector
```
Comment (ItayHoresh) [inline pipert/core/component.py:6]: The Splunk installation is optional. I think that this import will raise error if splunk is not installed.

Comment (ItayHoresh) [reply pipert/core/component.py:6]: @AmitKarin you need to look how the the pipe images and docker compose file working now and check this when splunk is not installed.

**Comment by ItayHoresh** [comment]
There is still implementation of splunk in pipe manager. It's need to be optional.

**Comment by AmitKarin** [comment]
@ItayHoresh all the checks are passed and i fixed the import of splunk.
can you approve?

**Pull request merged**